packages/react-components/lib/tasks/create-task.tsx

```diff
@@ -1246,6 +1277,51 @@ export function CreateTaskForm({
               />
             </Grid>
           </Grid>
+          <Grid container marginTop={theme.spacing(1)} marginLeft={theme.spacing(0)}>
+            <FormControl>
+              <FormLabel id="controlled-radio-buttons-group">Ends</FormLabel>
+              <RadioGroup
+                aria-labelledby="controlled-radio-buttons-group"
+                name="controlled-radio-buttons-group"
+                value={scheduleUntilValue}
+                onChange={handleScheduleUntilValue}
+              >
+                <FormControlLabel
+                  value={ScheduleUntilValue.NEVER}
+                  control={<Radio />}
+                  label="Never"
+                />
+                <Grid container direction="row" alignItems="center" spacing={2}>
+                  <Grid item xs={6}>
+                    <FormControlLabel
+                      value={ScheduleUntilValue.ON}
+                      control={<Radio />}
+                      label="On"
+                    />
+                  </Grid>
+                  <Grid item xs={6}>
+                    <DatePicker
+                      value={
+                        scheduleUntilValue === ScheduleUntilValue.NEVER
+                          ? new Date()
+                          : schedule.until
+                      }
+                      onChange={(date) =>
+                        date &&
+                        setSchedule((prev) => {
+                          date.setHours(23);
+                          date.setMinutes(59);
+                          return { ...prev, until: date };
+                        })
+                      }
+                      disabled={scheduleUntilValue !== ScheduleUntilValue.ON}
+                      renderInput={(props) => <TextField {...props} fullWidth />}
+                    />
+                  </Grid>
+                </Grid>
+              </RadioGroup>
+            </FormControl>
```

Comment:
```suggestion
           <FormControl fullWidth={true}>
              <FormHelperText>Ends</FormHelperText>
              <RadioGroup
                aria-labelledby="controlled-radio-buttons-group"
                name="controlled-radio-buttons-group"
                value={scheduleUntilValue}
                onChange={handleScheduleUntilValue}
                row
              >
                <Grid item xs={6} paddingLeft={theme.spacing(1)}>
                  <FormControlLabel
                    value={ScheduleUntilValue.NEVER}
                    control={<Radio />}
                    label="Never"
                  />
                </Grid>
                <Grid item xs={2} paddingLeft={theme.spacing(1)}>
                  <FormControlLabel
                    value={ScheduleUntilValue.ON}
                    control={<Radio />}
                    label="On"
                  />
                </Grid>
                <Grid item xs={4}>
                  <DatePicker
                    value={
                      scheduleUntilValue === ScheduleUntilValue.NEVER
                        ? new Date()
                        : schedule.until
                    }
                    onChange={(date) =>
                      date &&
                      setSchedule((prev) => {
                        date.setHours(23);
                        date.setMinutes(59);
                        return { ...prev, until: date };
                      })
                    }
                    disabled={scheduleUntilValue !== ScheduleUntilValue.ON}
                    renderInput={(props) => <TextField {...props} fullWidth />}
                  />
                </Grid>
              </RadioGroup>
            </FormControl>
```

I think we can use `FormHelperText` for the label, so it'll look similar to the labels of the other fields. My other suggestions here are mainly for using up empty space horizontally,

it'll look like this 
![image](https://github.com/open-rmf/rmf-web/assets/5383623/9879be10-4f1c-40ea-93ac-1736f9f3e924)

Reply:
This is great!
Thanks!
Done 8b8c8d85fc658518658664a1e610d8e0f72f3d00